sdk/storage/Azure.Storage.Files.Shares/tests/BenchmarkTests.cs

```diff
@@ -0,0 +1,114 @@
+        [TestCase(512 * Constants.MB)]
+        [TestCase(256 * Constants.MB)]
+        [TestCase(1024 * Constants.MB)]
+        public async Task TestMaxInitialDownload(long size)
+        {
+            var data = GetRandomBuffer(size);
+
+            await using DisposingFile test = await GetTestFileAsync();
+            ShareFileClient file = test.File;
+            await file.CreateAsync(size);
+            using (var stream = new MemoryStream(data))
+            {
+                await file.UploadAsync(stream);
+            }
+            Console.WriteLine("\n\n\nUpload successful.\n\n\n");
+            using (var resultStream = new MemoryStream(data))
+            {
+                await file.DownloadToAsync(resultStream, transferOptions: new StorageTransferOptions()
+                {
+                    InitialTransferSize = size,
+                    MaximumTransferSize = Constants.MB
+                });
+                Console.WriteLine("\n\n\nDownload successful.\n\n\n");
+                Assert.AreEqual(data.Length, resultStream.Length);
+                TestHelper.AssertSequenceEqual(data, resultStream.ToArray());
+            }
+        }
+
+        [Test]
+        [TestCase(512 * Constants.MB)]
+        [TestCase(256 * Constants.MB)]
+        [TestCase(1024 * Constants.MB)]
+        public async Task TestMaxSequentialDownload(long size)
+        {
+            var data = GetRandomBuffer(size + Constants.MB);
+
+            await using DisposingFile test = await GetTestFileAsync();
+            ShareFileClient file = test.File;
+            await file.CreateAsync(size + Constants.MB);
+            using (var stream = new MemoryStream(data))
+            {
+                await file.UploadAsync(stream);
+            }
+            Console.WriteLine("\n\n\nUpload successful.\n\n\n");
+            using (var resultStream = new MemoryStream(data))
+            {
+                await file.DownloadToAsync(resultStream, transferOptions: new StorageTransferOptions()
+                {
+                    InitialTransferSize = Constants.MB,
+                    MaximumTransferSize = size
+                });
+                Console.WriteLine("\n\n\nDownload successful.\n\n\n");
+                Assert.AreEqual(data.Length, resultStream.Length);
+                TestHelper.AssertSequenceEqual(data, resultStream.ToArray());
+            }
+        }
+
+        [Test]
+        [TestCase(512 * Constants.MB)]
+        [TestCase(256 * Constants.MB)]
+        public async Task TestDefaultInitialDownload(long size)
+        {
+            var data = GetRandomBuffer(size);
+
+            await using DisposingFile test = await GetTestFileAsync();
+            ShareFileClient file = test.File;
+            await file.CreateAsync(size);
+            using (var stream = new MemoryStream(data))
+            {
+                await file.UploadAsync(stream);
+            }
+            Console.WriteLine("\n\n\nUpload successful.\n\n\n");
```

Comment:
we should remove these writes to console from tests. assertions will tell us if they failed or not.